test/playwright/utils/navigation.ts

```diff
@@ -464,19 +464,12 @@ export const enableNewHeader = async (page: Page) => {
 
 export const enableOldHeader = async (page: Page) => {
   // Add the new_header cookie
-  await page.context().addCookies([
-    {
-      name: "features",
-      value: "%7B%22new_header%22%3A%22off%22%7D",
-      domain: "localhost",
-      path: "/",
-    },
-  ])
+  await setCookies(page.context(), { features: { new_header: "off" } })
 }
 
 export const setCookies = async (
   context: BrowserContext,
-  cookies: Record<string, string | boolean | string[]>
+  cookies: Record<string, string | boolean | string[] | Record<string, string>>
```

Comment:
Considering cookies are not limited in depth, it might be prudent to update the type to something more nesting-proof.

```ts
export interface CookieMap {
  [key: string]: string | boolean | string[] | CookieMap
}
```

Reply:
TIL: the types can be recursive :)